tests - housekeeping - logging

danlavu opened the pull request:
housekeeping, the following is looked at and may have been done:

* fixed typos and standardized formatting
* renamed test cases to improve the clarity of what the test does
* improved docstring language, setup, steps and expected results
* synced code with the docstring order
* removed necessary configuration relevant to the test
* added pytest.mark.importance to test cases
* added error messages to assertions

danlavu commented:
I am trying to figure out what the first test does; let's make it clear or remove it. 

```
42        if len(log_str.split("\n")) > 1:
43           assert "(0x3f7c0)" in log_str, f"String '{log_str}'WHAT DOES THIS EVEN DO?!"
```

jakub-vavra-cz commented on src/tests/system/tests/test_logging.py:
```diff
@@ -14,75 +17,74 @@
     :expectedresults:
-        1. "Starting with debug level = 0x0070" is in each file and
-            if log contains more than one line, log message with number "0x3f7c0" is stored
+        1. Logs messages contain default debug level
```
```suggestion
        1. Logs messages contain default debug level 0x0070
```

jakub-vavra-cz commented on src/tests/system/tests/test_logging.py:
```diff
@@ -14,75 +17,74 @@
-        1. Start SSSD with default debug level
-        2. Authenticate with user
-        3. Check that logs were not generated
+        1. Copy logs and login as user
+        2. Check logs again
```
```suggestion
        1. Store current logs and authenticate as a local user.
        2. Compare stored logs with the current ones.
```

jakub-vavra-cz commented on src/tests/system/tests/test_logging.py:
```diff
@@ -14,75 +17,74 @@
-        1. SSSD started successfully
-        2. User is authenticated
-        3. Diff of copy and logs is empty
+        1. Login successful
+        2. Before and after logs are empty
```
```suggestion
        1. Login was successful
        2. Login event did not generate any new log lines.
```

jakub-vavra-cz reviewed (commented):
Looks mostly ok, added a suggestion.

jakub-vavra-cz reviewed (approved):
OK

danlavu commented:
@alexey-tikhonov ping

alexey-tikhonov commented:
system tests are failing:
```
FAILED tests/test_logging.py::test_logging__default_settings_logs_sssd_conf_permissions_errors (client) - AssertionError: File check error messages are not in logs!
```

danlavu commented:
SSSD no longer requires explicit permission of '0600'. I updated the test to reflect that change. Unless this is not the expected behavior. I've parameterized other chmod values that do not contain read permissions to force the error.

alexey-tikhonov commented:
> SSSD no longer requires explicit permission of '0600'. I updated the test to reflect that change. Unless this is not the expected behavior. I've parameterized other chmod values that do not contain read permissions to force the error.

Previous value - 444 - has public 'read', so still should be failing

Btw, tests are still red.

Since we do not see `"SSSD erroneously started!"`, SSSD doesn't start (as expected). Maybe message format changed. Did you check the logs?

danlavu commented:
I did check the logs but I missed it the first time, it looks like the return code changed. I reverted the parameterization, since CI didn't like not being able to read sssd.conf and fixed the return code. It *should* work now.

alexey-tikhonov commented:
> it looks like the return code changed

It could, but why we didn't see `"SSSD erroneously started!"` message then?

alexey-tikhonov commented on src/tests/system/tests/test_logging.py:
```diff
@@ -91,70 +93,75 @@ def test_logging__default_debug_level_fatal_and_critical_failures(client: Client
     assert (
-        client.sssd.restart(debug_level=None, raise_on_error=False, apply_config=False).rc == 3
-    ), "SSSD didn't fail to read config, which is not expected"
+        client.sssd.start(debug_level=None, raise_on_error=False, apply_config=False).rc == 4
```
Let's check for `!= 0` (as in other places in this test).

alexey-tikhonov commented:
Test is still failing.

According to the test log, 'sssd' starts (when it shouldn't).

I just tried this on stock F41 and 'sssd' fails to start (as expected):
```
# ls -la /etc/sssd/sssd.conf 
-r--r--r--. 1 sssd sssd 352 May 31 15:42 /etc/sssd/sssd.conf

# systemctl start sssd
Job for sssd.service failed because the control process exited with error code.
See "systemctl status sssd.service" and "journalctl -xeu sssd.service" for details.

# journalctl -xeu sssd.service
sssd[3555]: [sssd] [sss_ini_read_sssd_conf] (0x0020): Permission check on config file>
sssd[3555]: Can't read config: 'File ownership and permissions check failed'
sssd[3555]: Failed to read configuration: 'File ownership and permissions check faile>
systemd[1]: sssd.service: Main process exited, code=exited, status=3/NOTIMPLEMENTED
```

alexey-tikhonov commented:
> According to the test log, 'sssd' starts (when it shouldn't).

No, if it would start, we would see `"SSSD erroneously started!"`

I think test just reads old logs, from a previous start.

danlavu commented:
> > According to the test log, 'sssd' starts (when it shouldn't).
> 
> No, if it would start, we would see `"SSSD erroneously started!"`
> 

You are right, I must have done something to get the wrong behavior. 

Ack, I will split the PR into two, and find an alternative to monitor to assert the error message.

danlavu commented:
Removed the test for now. 

https://github.com/SSSD/sssd/pull/7559 - is for 2-9 that includes the test.

alexey-tikhonov commented on src/tests/system/tests/test_logging.py:
```diff
@@ -14,147 +17,121 @@
     for file in [client.sssd.logs.monitor, client.sssd.logs.domain(), client.sssd.logs.nss, client.sssd.logs.pam]:
         log_str = client.fs.read(file)
-        assert "Starting with debug level = 0x0070" in log_str, f"Log file has wrong format: {log_str}"
+        assert "Starting with debug level = 0x0070" in log_str, "Debug level value are not in logs!"
```
I would read actual value and assert 'level == 0x0070'

alexey-tikhonov replied:
Exit message is still misleading.

If there will be "Starting with debug level = 0x0010" line in the log it is not  "Debug level value are not in logs!"
It is "Found unexpected value 0x0010 instead of 0x0070" as a log level.

alexey-tikhonov commented on src/tests/system/tests/test_logging.py:
```diff
@@ -14,147 +17,121 @@
+def test_logging__default_settings_logs_sbus_remove_key_error(client: Client):
     """
-    :title: SBUS doesn't trigger failure message at modules startup
+    :title: Default debug_level logs sbus remove key error
```
First, test actually checks that "does *not* log".

Seconds, I don't think that's worth automation at all, too narrow.
Maybe it would make more sense if it would check, for example, that there are no error messages at all (0x0010, 0x0020, 0x0040, 0x0080)
But check for a message specific to sbus impl (that very well could be changed meanwhile) isn't the best idea.

danlavu replied:
I don't like these logging tests, but other people seem to want them. I'll happily drop this test.

danlavu commented on src/tests/system/tests/test_logging.py:
```diff
@@ -14,147 +17,121 @@
 
         if len(log_str.split("\n")) > 1:
-            assert "(0x3f7c0)" in log_str, f"Log file has wrong format: {log_str}"
+            assert "(0x3f7c0)" in log_str, f"String '{log_str}'Debug level value are not in logs!"
```
It looks for a string that contains '(0x3f7c0)', everything after the ',' is an exit message.

alexey-tikhonov replied:
Exit message is wrong.

alexey-tikhonov replied:
@danlavu, exit message is still wrong.

Code checks for presence of "(0x3f7c0)" in the log (not sure what for, but ok) and if it's missing it prints "String ${entire_log_content} no in logs!"
What sense does it make?

alexey-tikhonov replied:
Once again - 0x3f7c0 is `SSSDBG_IMPORTANT_INFO` that is printed doesn't matter what 'debug_level' is set.

danlavu commented:
@alexey-tikhonov I made the same changes for the backport version, rebased and waiting for the tests to be green.

alexey-tikhonov commented:
Pushed PR: https://github.com/SSSD/sssd/pull/7496

* `master`
    * 0be58a26ee7feffbe03e0a817e7c0a6f2534a5e5 - tests - housekeeping - logging
